src/ghas_cli/utils/repositories.py

```diff
@@ -362,104 +362,65 @@ def get_languages(
+    codeql_aliased_languages = {
+        "typescript": "javascript",
+        "kotlin": "java",
+        "c#": "csharp",
+        "c++": "cpp",
+    }
 
     headers = network.get_github_headers(token)
-    languages = network.get(
+    languages_resp = network.get(
         url=f"https://api.github.com/repos/{organization}/{repository}/languages",
         headers=headers,
     )
-    if languages.status_code != 200:
+
+    if languages_resp.status_code != 200:
+        print(languages_resp.json())
         return ["default"]
 
-    lang = set()
-    for l in languages.json():
-        if only_interpreted:
-            if l.lower() in interpreted_languages:
-                lang.add(l.lower())
-            else:
-                if only_codeql:
-                    try:
-                        lang.add(aliased_interpreted_languages[l.lower()])
-                    except Exception as e:
-                        continue
+    languages = list()
+    for language in [l.lower() for l in languages_resp.json()]:
+        if only_codeql:
+            if language in codeql_languages:
+                languages.append(language)
+            elif language in codeql_aliased_languages:
+                languages.append(codeql_aliased_languages[language])
         else:
-            lang.add(l.lower())
+            languages.append(language)
 
-    if not lang:
-        return ["default"]
-    else:
-        return list(lang)
+    return languages
 
 
-def load_codeql_base64_template(language: str, default_branch: str = "main") -> tuple:
-    language = language.lower()
-    try:
-        with open(f"./templates/codeql-analysis-{language.lower()}.yml", "r") as f:
-            # Ugly af but `yaml` transforms `on:` to `True:` which is obviously annoying to parse GitHub Actions files..
-            template = f.readlines()
-            template_new = ""
-            for l in template:
-                if l == '    branches: ["main"]\n':
-                    template_new += f"    branches: ['{default_branch}']\n"
-                else:
-                    template_new += l
-    except Exception as e:
-        with open(f"./templates/codeql-analysis-default.yml", "r") as f:
-            language = "default"
-            template = f.readlines()
-            template_new = ""
-            for l in template:
-                if l == '    branches: ["main"]\n':
-                    template_new += f"    branches: ['{default_branch}']\n"
-                else:
-                    template_new += l
-    return language, str(
-        base64.b64encode(template_new.encode(encoding="utf-8")), "utf-8"
-    )
+def load_codeql_base64_template(
+    languages: List, default_branch: List = ["main"]
+) -> str:
+    with open(f"./templates/codeql-analysis-default.yml", "r") as f:
+        data = "".join(f.readlines())
+        data = data.replace("""branches: [ ]""", f"""branches: {default_branch}""")
```

Comment:
I think it would be great to have the scans triggered on PRs against any branches, and on any commits against the default branch. We can leave it as the latter for now, but I think it's something we can discuss for a future release?

Reply:
Yes that's a hassle :/ We can discuss it more later.